README.md

```diff
@@ -1,35 +1,251 @@
+
+In this tutorial, we explain how you can integrate an analysis tool of your choice in Codacy.
+You can check the code of an already implemented tool and, if you wish, fork it to start your own.
+You are free to modify and use it for your own tools.
+
+#### Requirements
+
+* Docker definition with the tool you want to integrate
+* Define the documentation for the patterns provided by the tool
+
+#### Assumptions and Behaviour
+
+* To run the tool we provide the configuration file, `/.codacyrc`, with the language to run and optional parameters your tool might need.
+* The source code to be analysed will be located in `/src`, meaning that when provided in the configuration, the file paths are relative to `/src`.
+
+* **.codacyrc**
+  * **files:** Files to be analysed (their path is relative to `/src`)
+  * **tools:** Array of tools
+    * **name:** Unique identifier of the tool
+    * **patterns:** Array of patterns that must be checked
+      * **patternId:** Unique identifier of the pattern
+      * **parameters:** Parameters of the pattern
+        * **name:** Unique identifier of the parameter
+        * **value:** Value to be used as parameter value
+
+```json
+{
+  "files" : ["foo/bar/baz.js", "foo2/bar/baz.php"],
+  "tools":[
+    {
+      "name":"jshint",
+      "patterns":[
+        {
+          "patternId":"latedef",
+          "parameters":[
+            {
+              "name":"latedef",
+              "value":"vars"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+Regarding to the configuration file, the tool should have different behaviours for the following situations:
+* If /.codacyrc exists and has files and patterns, use them to run.
+* If /.codacyrc exists and has only files and no patterns, run only for those files.
+* If /.codacyrc does not exist or any of its contents (files or patterns) is not available, 
+you should invoke the tool for all files from /src (files should be searched recursively for all folders in /src) 
+and check them with the default patterns.
+* If /.codacyrc fails to be parsed, throw an error.
+
+**Exit codes**
+* The exit codes can be different, depending if the tool invocation is successful or not:
+  * :tada: **0**: The tool executed successfully
+  * :cold_sweat: **1**: An unknown error occurred while running the tool 
+  * :alarm_clock: **2**: Execution timeout
+
+**Notes:**
+
+* To run the tool in debug mode, so you can have more detailed logs, you need to set the environment variable `DEBUG` to `true` when invoking the docker.
+* To configure a different timeout for the tool, you have to set the environment variable `TIMEOUT` when invoking the docker, setting it with values like `10 seconds`, `30 minutes` or `2 hours`.
+
+## Setup
+1. Write the docker file that will run the tool.
+   * It must have a binary entry point without any parameters.
```

Comment:
Either that, or use this project, which does this setup for you. Maybe split this into two sections (if using or not using the seed).

Reply:
As we previously talked, the seed doesn't provide anything related with the creation of the dockerfile. Therefore, I added a reference to the sbt-native-packager plugin, that can be used if the contributor uses this seed, as if he will be using scala and probably, sbt.